Packs/Workday/Integrations/WorkdaySignOnEventCollector/WorkdaySignOnEventCollector.py

```diff
@@ -0,0 +1,565 @@
+        Tests API connectivity and authentication.
+
+        :return: 'ok' if test passed, else exception.
+        :rtype: ``str``
+        """
+        seconds_ago = 5
+        from_time = get_from_time(seconds_ago)
+        to_time = datetime.now(tz=timezone.utc).strftime(DATE_FORMAT)
+
+        payload = self.generate_test_payload(from_time=from_time, to_time=to_time)
+
+        self._http_request(
+            method="POST", url_suffix="", data=payload, resp_type="text", timeout=120
+        )
+
+        return "ok"
+
+
+""" HELPER FUNCTIONS """
+
+
+def convert_to_json(response: str | dict) -> tuple[Dict[str, Any], Dict[str, Any]]:
+    """
+    Convert an XML response to a JSON object and extract the 'Workday_Account_Signons' data.
+
+    :param response: XML response to be converted
+    :return: Tuple containing the full converted response and the extracted 'Workday_Account_Signons' data.
+    :raises ValueError: If the expected data cannot be found in the response.
+    """
+    if type(response) == dict:
+        raw_json_response = response
+    else:
+        try:
+            raw_json_response = json.loads(xml2json(response))
+        except Exception as e:
+            raise ValueError(f"Error parsing XML to JSON: {e}")
+
+    # Get the 'Get_Workday_Account_Signons_Response' dictionary safely
+    response_data = demisto.get(raw_json_response, "Envelope.Body.Get_Workday_Account_Signons_Response")
+
+    if not response_data:
+        response_data = raw_json_response.get(
+            "Get_Workday_Account_Signons_Response", {}
+        )
+
+    account_signon_data = response_data.get("Response_Data", {})
+
+    # Ensure 'Workday_Account_Signon' is a list
+    workday_account_signons = account_signon_data.get("Workday_Account_Signon")
+    if isinstance(workday_account_signons, dict):
+        account_signon_data["Workday_Account_Signon"] = [workday_account_signons]
+
+    return raw_json_response, account_signon_data
+
+
+def process_and_filter_events(events: list, from_time: str, previous_run_checksums: set) -> tuple:
+    non_duplicates = []
+    future_potential_duplicates = []
+    checksums_for_next_iteration = set()
+
+    from_datetime = datetime.strptime(from_time, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
+    most_recent_event_time = datetime.min.replace(tzinfo=timezone.utc)
+
+    for event in events:
+        event_datetime = datetime.strptime(event["Signon_DateTime"], DATE_FORMAT).replace(tzinfo=timezone.utc)
+
+        # Add '_time' key to each event
+        event["_time"] = event.get("Signon_DateTime")
+
+        # Update the most recent event time
+        if event_datetime > most_recent_event_time:
+            most_recent_event_time = event_datetime
+
+        # Check for duplicates within ±1 second of from_time
+        if abs((event_datetime - from_datetime).total_seconds()) <= 1:
+            event_checksum = generate_checksum(event)
+            if event_checksum not in previous_run_checksums:
```

Comment:
lets save in a list all the IDs that has been filtered and log the amount of them and the IDs(in separate log) in the end of the loop. don't print if the list is empty.